engine/src/engine.rs

```diff
@@ -1386,8 +1386,18 @@ where
                     Some(log.into())
                 }
             } else if log.address == cross_contract_call::ADDRESS.raw() {
-                if let Ok(promise) = PromiseCreateArgs::try_from_slice(&log.data) {
-                    crate::xcc::handle_precompile_promise(io, handler, promise, current_account_id);
+                if log.topics[0] == cross_contract_call::AMOUNT_TOPIC {
+                    let required_near =
+                        Yocto::new(U256::from_big_endian(log.topics[1].as_bytes()).low_u128());
```

Comment:
Using `low_u128` here looks like a red flag. Similar to down-casting using `as`!

Should we check that higher bytes are `0`?

Anyway, from the calling site, I see that higher bytes will always be zero, so not sure this is worth it, but just highlighting it here :)

Reply:
Added a check in https://github.com/aurora-is-near/aurora-engine/pull/572/commits/e51aec8fdd4d601e5f7829e0fad50827eb78863f